### PR TITLE
feat: Add requesting storage persistence for more quota

### PR DESCRIPTION
**Pull request by fl0rek**
related #317 

[MDN page](https://developer.mozilla.org/en-US/docs/Web/API/Storage_API/Storage_quotas_and_eviction_criteria) isn't exact on the details (as detail vary by browser), but requesting persistent storage on firefox raises the quota limit (and eliminates quota issues on my machine).
Browsers should also try more to keep our data from being removed.

Node that this will display prompt on Firefox, while chromium based browsers use heuristics to figure out whether to grant the storage.

image

### CODE REVIEW

File: node-wasm/src/node.rs [hunk 1]
```diff
@@ -92,6 +93,10 @@ impl NodeDriver {
         worker_script_url: &str,
         worker_type: Option<NodeWorkerKind>,
     ) -> Result<NodeDriver> {
+        if let Err(e) = request_storage_persistence().await {
```
Comment (zvolin) [inline node-wasm/src/node.rs:96]: since this only affects quota in ff, can we guard this to be only called in ff?

Comment (fl0rek) [reply node-wasm/src/node.rs:96]: I could have mentioned that beforehand, but

> In Firefox, when a site chooses to use persistent storage, the user is notified with a UI popup that their permission is requested.
> Safari and most Chromium-based browsers, such as Chrome or Edge, automatically approve or deny the request based on the user's history of interaction with the site and do not show any prompts to the user.

So the prompt would be displayed on firefox, other browser doing some heuristics. Maybe just leave it for all browsers then? We are intending on manually managing the data anyway and I want to do as little parsing of UserAgent as possible. Sniffing for chrome already felt bad enough :P

File: node-wasm/src/error.rs [hunk 2]
```diff
@@ -50,6 +50,16 @@ impl Error {
     }
 }
 
+impl Display for Error {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        if let Some(error) = self.0.dyn_ref::<js_sys::Error>() {
+            write!(f, "{} ({})", error.name(), error.message())
```
Comment (zvolin) [inline node-wasm/src/error.rs:56]: could you also add a new line with the `.cause()`?

**Review by oblique — APPROVED** [review]
LGTM 🙈

File: node-wasm/src/error.rs (hunk 2 above)
Comment (fl0rek) [reply node-wasm/src/error.rs]: Good idea, added printing `cause`s in a loop until we get to the bottom of the error 👍

**Review by oblique — APPROVED** [review]

**Review by zvolin — APPROVED** [review]

**Pull request merged**